### ISSUE TITLE
[feature request] add in encrypt/decrypt method for common/keyring

### ISSUE THREAD

**Issue by RoyTimes** [polkadot-js/common#1068]
Hi, per discussion in https://github.com/polkadot-js/common/issues/1067 The project we are working on needs to either obtain private key from browser extension **OR** enable the browser extension to encrypt/decrypt message with a recipient's public key and the private key from the keyring. 

I guess it's better not to mess with exposing the private key. Therefore, I guess the second option would be optimal. 

A few questions before we start drafting a PR:
1. Do you think encrypt/decrypt are good functionalities to be included? I believe Metamask offers some ways of doing so. 
2. "It might be tricky to add in support for all elliptic curves: ed25519 is covered by tweetnacl and we work with the lib a lot; secp256k1 is implemented by "elliptic" and offers an ECDH and should be good; sr25519 is tricky and I wonder if I should mess with the Rust wasm code. It's good enough for us to only have ed25519, or use ed25519 for all encrypt/decrypt. I wonder how much compromise is acceptable. " (EDITED: I might be foggy on this. Both sr25519 & ed25519 should use the same curve for ECDH)
3. For browser extension, we will also propose a PR related to this but won't open a new issue. 
4. For browser extension, I wonder if it should pop up a new window similar to how "signing" works, or the extension could do it quietly. I don't think there's much risk for user to encrypt data (unlike signing), but might be required to pop up a window when decrypting. 

Let me know your thoughts. Tests will be included.

**Comment by jacogr** [comment]
Closed in #1070
